src/lib/functions/registry.js

```diff
@@ -1,15 +1,19 @@
 // @ts-check
-const { mkdir } = require('fs').promises
-const { env } = require('process')
+import { promises } from 'fs'
+import { env } from 'process'
 
-const { NETLIFYDEVERR, NETLIFYDEVLOG, chalk, log, warn } = require('../../utils')
-const { getLogMessage } = require('../log')
+import { listFunctions } from '@netlify/zip-it-and-ship-it'
```

Comment:
@ehmicky or @erezrokah maybe you have a better idea but the require was in the constructor for performance reasons.

I don't know any better way that does not require huge refactorings as a constructor cannot be async or an unawaited promise can cause race conditions.

Do you know how large the performance impact will be with that?

Reply:
Could this be moved out of the constructor?
For example:

```js
const zipItAndShipItPromise = import('@netlify/zip-it-and-ship-it')

class FunctionsRegistry {
  // Does not set `listFunctions` anymore
  constructor(...) {...}
  ...

  async scan(directories) {
    const { listFunctions } = await zipItAndShipItPromise
    const functions = listFunctions()
  },
```

On my machine, ZISI takes 500ms to load. However, it is using some modules shared by other CLI dependencies, so some of those 500ms are spent loading modules which would need to be loaded anyway.

Still, 500ms is somewhat significant, so we might want to keep it this performance optimization, since it is fairly simple to do so. What do you think?

Also cc @eduardoboucas who probably introduced this in https://github.com/netlify/cli/pull/2826

Reply:
I believe it was me do did the optimization in https://github.com/netlify/cli/pull/2585, so it can have a big difference.
I recommend running some benchmarks to verify.